src/main/java/com/internship/frejaeidjmeterplugin/jmeter/sampler/GeneralSampler.java

```diff
@@ -0,0 +1,90 @@
+package com.internship.frejaeidjmeterplugin.jmeter.sampler;
+
+import com.internship.frejaeidjmeterplugin.jmeter.frejaRequests.AuthenticationService;
+import com.internship.frejaeidjmeterplugin.jmeter.frejaRequests.SignService;
+import com.verisec.frejaeid.client.beans.authentication.get.AuthenticationResult;
+import com.verisec.frejaeid.client.beans.sign.get.SignResult;
+import com.verisec.frejaeid.client.enums.MinRegistrationLevel;
+import com.verisec.frejaeid.client.exceptions.FrejaEidClientInternalException;
+import com.verisec.frejaeid.client.exceptions.FrejaEidClientPollingException;
+import com.verisec.frejaeid.client.exceptions.FrejaEidException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.jmeter.samplers.AbstractSampler;
+import org.apache.jmeter.samplers.Entry;
+import org.apache.jmeter.samplers.SampleResult;
+
+public class GeneralSampler extends AbstractSampler {
+
+    private final AuthenticationService authService;
+    private final SignService signService;
+
+    public GeneralSampler() throws FrejaEidClientInternalException {
+        authService = new AuthenticationService();
+        signService = new SignService();
+    }
+
+    public String getEmail() {
+        return getPropertyAsString("email");
+    }
+
+    public void setEmail(String email) {
+        setProperty("email", email);
+    }
+
+    public void setSelected(String text) {
+        setProperty("selected", text);
+    }
+
+    public String getSelected() {
+        return getPropertyAsString("selected");
+    }
+
+    @Override
+    public SampleResult sample(Entry entry) {
+        SampleResult sr = new SampleResult();
+
+        try {
+            if (getSelected().equals("auth")) {
```

Comment:
Maybe it would be more readable to write this piece of code with switch statement.